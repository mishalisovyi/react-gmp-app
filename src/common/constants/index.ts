export const Icon = {
  VERTICAL_ELLIPSIS: String.fromCharCode(8942),
  MULTIPLICATION_X: String.fromCharCode(10005),
  TELEPHONE_RECORDER: String.fromCharCode(8981),
};

export const API_URL = 'http://localhost:4000';

export const HTTP_RESPONSE_STATUS_NO_CONTENT = 204;
