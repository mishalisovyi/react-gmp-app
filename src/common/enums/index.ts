export { ButtonType } from './button-type.enum';
export { HttpMethod } from './http-method.enum';
