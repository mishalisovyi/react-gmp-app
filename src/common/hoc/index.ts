export { withPortalRendering } from './withPortalRendering/withPortalRendering';
export { ModalWindow } from './ModalWindow/ModalWindow';
