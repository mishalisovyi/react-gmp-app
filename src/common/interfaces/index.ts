export { EnumerableComponentProps } from './EnumerableComponentProps';
export { ModalWindowProps, ModalWindowActionButtonProps } from './ModalWindowProps';
export { SelectListData, SelectListValue } from './SelectListData';
