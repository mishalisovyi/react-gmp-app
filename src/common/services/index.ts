export {
  sendGetRequest,
  sendPostRequest,
  sendPutRequest,
  sendDeleteRequest,
} from './http.service';
