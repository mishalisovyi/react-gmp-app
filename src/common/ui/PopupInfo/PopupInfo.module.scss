@use '../../../colors';
@use '../../../mixins';

.PopupInfo {
  text-align: center;

  &__check-mark {
    @include mixins.typography-body-large;

    display: flex;
    justify-content: center;
    align-items: center;
    width: 60px;
    height: 60px;
    margin: 32px auto;
    border-radius: 50%;
    background-color: colors.$festive-fennec;
  }
}