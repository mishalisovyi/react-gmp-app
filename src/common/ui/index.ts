export { Button } from './Button/Button';
export { Divider } from './Divider/Divider';
export { DropdownMenu } from './DropdownMenu/DropdownMenu';
export { FilterSelectList } from './FilterSelectList/FilterSelectList';
export { LabelPrimary } from './LabelPrimary/LabelPrimary';
export { LabelSecondary } from './LabelSecondary/LabelSecondary';
export { Logo } from './Logo/Logo';
export { PopupConfirmation } from './PopupConfirmation/PopupConfirmation';
export { PopupDialog } from './PopupDialog/PopupDialog';
export { PopupInfo } from './PopupInfo/PopupInfo';
export { SearchPanel } from './SearchPanel/SearchPanel';
export { Spinner } from './Spinner/Spinner';
export { Tab } from './TabBar/Tab/Tab';
export { TabBar } from './TabBar/TabBar';
export { TextField } from './TextField/TextField';
