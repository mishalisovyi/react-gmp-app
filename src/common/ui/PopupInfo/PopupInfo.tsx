import React, { MouseEventHandler } from 'react';

import { ModalWindow } from 'common/hoc';

import styles from './PopupInfo.module.scss';

interface PopupInfoProps {
  title: string;
  message: string;
  onClose: MouseEventHandler<HTMLButtonElement>;
}

export function PopupInfo({ title, message, onClose }: PopupInfoProps) {
  return (
    <ModalWindow onClose={onClose}>
      <div className={styles['PopupInfo']}>
        <div className={styles['PopupInfo__check-mark']}>&#10003;</div>
        <h1 className="typography-heading">{title}</h1>
        <p className="typography-body">{message}</p>
      </div>
    </ModalWindow>
  );
}
