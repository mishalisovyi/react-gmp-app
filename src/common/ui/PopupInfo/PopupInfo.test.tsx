import React, { ReactNode, ReactPortal } from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Icon } from 'common/constants';

import { PopupInfo } from './PopupInfo';

const handlePopupClosing = jest.fn();

beforeAll(() => {
  ReactDOM.createPortal = (node: ReactNode): ReactPortal => node as ReactPortal;
});

it('should render info popup', () => {
  const tree = renderer.create(<PopupInfo title="test title" message="test message" onClose={handlePopupClosing} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should handle info popup closing', async () => {
  render(<PopupInfo title="test title" message="test message" onClose={handlePopupClosing} />);

  await userEvent.click(screen.getByText(Icon.MULTIPLICATION_X));

  expect(handlePopupClosing).toHaveBeenCalled();
});
