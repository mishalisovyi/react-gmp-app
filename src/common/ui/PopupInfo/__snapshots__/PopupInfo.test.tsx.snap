// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render info popup 1`] = `
<div>
  <div />
  <div>
    <div>
      <button
        className="button button--transparent"
        onClick={[MockFunction]}
        type="button"
      >
        ✕
      </button>
    </div>
    <div>
      <div>
        <div>
          ✓
        </div>
        <h1
          className="typography-heading"
        >
          test title
        </h1>
        <p
          className="typography-body"
        >
          test message
        </p>
      </div>
    </div>
    <div />
  </div>
</div>
`;
